### PR TITLE
[DOCS] Reformats cluster allocation explain API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45197.

This PR updates the cluster allocation explain API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Cluster allocation explain API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/cluster.allocation_explain.json)
* [Cluster allocation explain API doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-allocation-explain.html)

### CODE REVIEW

File: docs/reference/cluster/allocation-explain.asciidoc [hunk 1]
```diff
@@ -1,28 +1,65 @@
...
+    explanation for.
+
+You can also have {es} explain the allocation of the first unassigned shard that 
+it finds by sending an empty body for the request.
+
+
+[[cluster-allocation-explain-api-examples]]
+==== {api-examples-title}
 
 [source,js]
```
Comment (jrodewig) [inline docs/reference/cluster/allocation-explain.asciidoc]: This example requires the `myindex` index to exist. The [previous example](https://github.com/elastic/elasticsearch/pull/45603/files#diff-c13719368a87e7bf448d3765e1593f0aL18) included that.

If you don't want to explicitly set up the index, you can do it inside a comment like this:
```suggestion

//////
[source,js]	
--------------------------------------------------	
PUT /myindex	
--------------------------------------------------	
// CONSOLE	
// TESTSETUP
//////

[source,js]
```

Once that's in there, it should fix the build issues.

**Review by jrodewig — CHANGES_REQUESTED** [review]
Left a suggestion to help fix the build CI.

File: docs/reference/cluster/allocation-explain.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/cluster/allocation-explain.asciidoc]: @jrodewig Wow! Thank you for the help!

**Review by jrodewig — APPROVED** [review]
LGTM

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**